File: java/doublylinkedlist.java
class node {
    int value;
    node next, prev;

    public node(int value) {
        this.value = value;
        this.next = null;
        this.prev = null;
    }
}

public class doublylinkedlist {
    node head, tail;

    public doublylinkedlist() {
        this.head = null;
        this.tail = null;
    }

    void traverse(node head) {
        node current = head;
        while (current != null) {
            System.out.print(current.value + " ");
            current = current.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        node head = new node(1);
        node second = new node(0);
        node third = new node(7);

        head.next = second;
        second.next = third;
        third.next = null;
        third.prev = second;
        second.prev = head;
        head.prev = null;

        doublylinkedlist d = new doublylinkedlist();
        d.traverse(head);
    }
  
}

File: doublylinkedlist.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
        self.prev = None

    def get_next(self):
        return self.next
    
    def get_prev(self):
        return self.prev
    
    def set_next(self, new_next):
        self.next = new_next

    def set_prev(self, new_prev):
        self.prev = new_prev

    def __str__(self):
        return f"prev: {self.prev}, data: {self.data}, next: {self.next}"
    

class DoublyLinkedList:

    def __init__(self):
        self.head = None
        self.tail = None

    def set_head(self, new_head_node):
        self.head = new_head_node.data

    def set_tail(self, new_tail_node):
        self.tail = new_tail_node.data

    def forward_traversal(self):
        current = self.head

        while current is not None:
            print(current.data)
            current = current.next

    def backward_traversal(self):
        current = self.tail

        while current is not None:
            print(current.data)
            current = current.prev

    def get_len(self):
        count = 0
        current = self.head

        while current is not None:
            count += 1
            current = current.next

        return count
    
    def insert(self, new_node, prev_node):
        pass


    